pkg/bootstrap/os/tasks.go

```diff
@@ -398,13 +398,28 @@ type InstallPackage struct {
 
 func (i *InstallPackage) Execute(runtime connector.Runtime) error {
 	host := runtime.RemoteHost()
+	release, ok := host.GetCache().Get(Release)
+	if !ok {
+		return errors.New("get os release failed by host cache")
+	}
+	os := release.(*osrelease.Data).ID
+	var pkg []string
+	switch strings.ToLower(os) {
+	case "ubuntu", "debian":
+		pkg = i.KubeConf.Cluster.System.Debs
+	case "centos", "rhel":
+		pkg = i.KubeConf.Cluster.System.Rpms
+	default:
+		return fmt.Errorf("unsupported operation system %s", os)
+	}
```

Comment:
I guess that would not work when the os distribution is Kylin, fedora, or others. So I suggest using assert to check the repo type, and we can use this as follow:
```go
repo, ok := host.GetCache().Get("repo")
if !ok {
	return errors.New("get repo failed by host cache")
}
r := repo.(repository.Interface)

var pkg []string
if _, ok := r.(*repository.Debian); ok {
  pkg = i.KubeConf.Cluster.System.Debs
} else if _, ok := r.(*repository.RedhatPackageManager); ok {
  pkg = i.KubeConf.Cluster.System.Rpms
}
```

Reply:
thanks for your prompt reply.